src/shared/sleep-config.c

```diff
@@ -108,77 +111,162 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
+/* Get the list of battery */
+static int get_sd_device_battery(sd_device_enumerator **ret) {
+        int r;
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+
+        r = sd_device_enumerator_new(&e);
+        if (r < 0)
+                return log_oom();
+
+        r = sd_device_enumerator_allow_uninitialized(e);
+        if (r < 0)
+                return r;
+
+        r = sd_device_enumerator_add_match_subsystem(e, "power_supply", true);
+        if (r < 0)
+                return r;
+
+        r = sd_device_enumerator_add_match_property(e, "POWER_SUPPLY_TYPE", "Battery");
+        /* No battery found check */
+        if (r < 0)
+                return log_error_errno(r, "Failed to get property match to battery: %m");
+
+        *ret = TAKE_PTR(e);
+        return 0;
+}
+
 /* If battery percentage capacity is less than equal to 5% return success */
 int battery_is_low(void) {
-        int r;
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        sd_device *dev;
+        int c = 0, r;
 
          /* We have not used battery capacity_level since value is set to full
          * or Normal in case acpi is not working properly. In case of no battery
          * 0 will be returned and system will be suspended for 1st cycle then hibernated */
 
-        r = read_battery_capacity_percentage();
-        if (r == -ENOENT)
-               return false;
+        r = get_sd_device_battery(&e);
         if (r < 0)
-               return r;
+                return log_error_errno(r, "Failed to get available battery: %m");
 
-        return r <= 5;
+        FOREACH_DEVICE(e, dev) {
+                r = read_battery_capacity_percentage(dev);
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to get battery capacity: %m");
+                        continue;
+                }
+                if (r > 5)
+                        /* increment only if battery capacity is above 5 */
+                        c++;
+        }
+        /* if all battery are below 5 then c will remain 0*/
+        if (c == 0)
```

Comment:
given it's used as a true/false it doesn't need to be a counter, just use a bool

Reply:
yeah. right. fixed